static/js/src/cube/components/CubeHeader/Hero.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { useEffect, useRef, useState } from "react";
+import { Col, Row } from "@canonical/react-components";
+import { Module } from "../../types";
+
+type Props = {
+  modules: Module[];
+  isLoading: boolean;
+  certifiedBadge: null | Record<string, string>;
+  hasEnrollments: boolean;
+};
+
+const Hero = ({
+  modules,
+  isLoading,
+  certifiedBadge,
+  hasEnrollments,
+}: Props) => {
+  const [heroStripIsShallow, setHeroStripIsShallow] = useState(false);
+  const [heroContentMarginTop, setHeroContentMarginTop] = useState("2rem");
+  const heroStrip = useRef<HTMLElement>(null);
+
+  useEffect(() => {
+    const resizeObserver = new ResizeObserver((entries) => {
+      entries.forEach((entry) => {
+        const width = Math.floor(entry.contentRect.width);
+        if (width < 1035) {
+          setHeroStripIsShallow(false);
+          setHeroContentMarginTop("2rem");
+        } else {
+          setHeroStripIsShallow(true);
+          setHeroContentMarginTop("-2rem");
+        }
+      });
+    });
+
+    if (heroStrip.current) {
+      resizeObserver.observe(heroStrip.current);
+    }
+  }, [heroStrip.current]);
```

Comment:
What does this do? Could this be done via CSS media queries?

Reply:
This prevents the hero section from getting vertically squashed on smaller screens.

Setting the margin could definitely be done with a media query, but I'm not sure about applying the `is-shallow` class. I'll try to replace with a media query.

For reference, [the old code](https://github.com/canonical-web-and-design/ubuntu.com/pull/10953/files#diff-8e9f6cac8e2dd10b76e913ecb65ad88adbad037ec9d084a27cf3e506d53f8e56L215) did this with some javascript in the template.

Reply:
I changed this to use CSS with a media query.